packages/sanity/src/core/studio/StudioProvider.tsx

```diff
@@ -70,12 +71,14 @@ export function StudioProvider({
     <WorkspaceLoader LoadingComponent={LoadingBlock} ConfigErrorsComponent={ConfigErrorsScreen}>
       <StudioTelemetryProvider config={config}>
         <LocaleProvider>
-          <VersionProvider>
-            <PackageVersionStatusProvider>
-              <EnableErrorReporting />
-              <ResourceCacheProvider>{children}</ResourceCacheProvider>
-            </PackageVersionStatusProvider>
-          </VersionProvider>
+          <AddonDatasetProvider>
```

Comment:
Had to add it in order to use `useBundlesStore` in the context.
Will likely remove it once I have the updates to the bundle store -> context

Reply:
My guess is that depending on how we go about this: no we will likely not need the version provider at all if the releases plugin is not added to the studio.

(hence why I haven't really spent much time in the contextPlugin and looks messy 🫠 )

However, because of we are right now development wise, we do need something like the version context at a higher level because the currentVersion (or the version that is picked is across multiple levels in the studio: navbar, document level and footer in the document)

Reply:
If we move it to the releases plugin, the context will be added at the studio level, so it will be possible to use it from the navbar, document and footer.

We could have in `packages/sanity/src/core/releases/plugin/ReleasesStudioLayout.tsx`

```ts
import {VersionProvider} from '../../../_singletons/core/form/VersionContext'
import {type LayoutProps} from '../../config'
import {BundlesProvider} from '../../store/bundles/BundlesProvider'
import {AddonDatasetProvider} from '../../studio'

export function ReleasesStudioLayout(props: LayoutProps) {
  // TODO: Replace for useReleasesEnabled
  const {enabled, mode} = {enabled: true, mode: 'default'}

  if (!enabled) {
    return props.renderDefault(props)
  }

  return (
    <AddonDatasetProvider>
      <BundlesProvider>
        <VersionProvider>{props.renderDefault(props)}</VersionProvider>
      </BundlesProvider>
    </AddonDatasetProvider>
  )
}
```

And then, in the `<VersionProvider />` we can have access to the bundlesProvider and use `useBundles` instead of `useBundlesStore` [here](https://github.com/sanity-io/sanity/blob/corel-35/packages/sanity/src/_singletons/core/form/VersionContext.tsx#L31-L32).

Reply:
This makes a lot of sense, if you're ok with it I'll make this change in a separate PR since I'm already focusing on updating the VersionProvider in a different branch :)

Reply:
This component will add the necessary providers here https://github.com/sanity-io/sanity/blob/corel-35/packages/sanity/src/core/studio/Studio.tsx#L117-L118

So it's wrapping the studio tool.

Reply:
Sure @RitaDias makes sense!

Reply:
Related to the conversation of the context & hooks, this will likely be removed in a separate PR :)